src/libostree/ostree-repo-composefs.c

```diff
@@ -322,30 +322,37 @@ checkout_one_composefs_file_at (OstreeRepo *repo, const char *checksum, struct l
 
       guchar *known_digest = NULL;
 
-#ifdef HAVE_LINUX_FSVERITY_H
-      /* First try to get the digest directly from the bare repo file.
-       * This is the typical case when we're pulled into the target
-       * system repo with verity on and are recreating the composefs
-       * image during deploy. */
-      char buf[sizeof (struct fsverity_digest) + OSTREE_SHA256_DIGEST_LEN];
-
-      if (G_IS_UNIX_INPUT_STREAM (input))
+      if (verity != OT_TRISTATE_NO)
         {
-          int content_fd = g_unix_input_stream_get_fd (G_UNIX_INPUT_STREAM (input));
-          struct fsverity_digest *d = (struct fsverity_digest *)&buf;
-          d->digest_size = OSTREE_SHA256_DIGEST_LEN;
-
-          if (ioctl (content_fd, FS_IOC_MEASURE_VERITY, d) == 0
-              && d->digest_size == OSTREE_SHA256_DIGEST_LEN
-              && d->digest_algorithm == FS_VERITY_HASH_ALG_SHA256)
-            known_digest = d->digest;
-        }
+#ifdef HAVE_LINUX_FSVERITY_H
+          /* First try to get the digest directly from the bare repo file.
+           * This is the typical case when we're pulled into the target
+           * system repo with verity on and are recreating the composefs
+           * image during deploy. */
+          char buf[sizeof (struct fsverity_digest) + OSTREE_SHA256_DIGEST_LEN];
```

Comment:
Can you move this into the block with the `ioctl`?  It has no meaning outside of that block.

Reply:
Oh.  I see.  `known_digest` potentially ends up pointing into this buffer.  Then *definitely* bring `known_digest` in to this block to reduce its scope to the same scope as `buf`.  In the original code they were at the same logical level (although one of them was inside an `#ifdef` and the other not).  After the change, they are in two separate `block { }s` which is definitely suspect.

Reply:
Definitely good to fix, though since the variable wasn't *used* outside that inner block I don't think there's anything suspect in practice right now.